stl/inc/generator

```diff
@@ -71,23 +87,23 @@ public:
     }
 
     template <class _Alloc2, class... _Args>
-        requires convertible_to<const _Alloc2&, _Allocator>
+        requires convertible_to<const _Alloc2&, _Proto_allocator>
     static void* operator new(const size_t _Size, allocator_arg_t, const _Alloc2& _Al, const _Args&...) {
-        return _Allocate(static_cast<_Alloc>(static_cast<_Allocator>(_Al)), _Size);
+        return _Allocate(static_cast<_Alloc>(static_cast<_Proto_allocator>(_Al)), _Size);
     }
 
     template <class _This, class _Alloc2, class... _Args>
-        requires convertible_to<const _Alloc2&, _Allocator>
+        requires convertible_to<const _Alloc2&, _Proto_allocator>
     static void* operator new(const size_t _Size, const _This&, allocator_arg_t, const _Alloc2& _Al, const _Args&...) {
-        return _Allocate(static_cast<_Alloc>(static_cast<_Allocator>(_Al)), _Size);
+        return _Allocate(static_cast<_Alloc>(static_cast<_Proto_allocator>(_Al)), _Size);
     }
 
     static void operator delete(void* const _Ptr, const size_t _Size) noexcept {
         if constexpr (default_initializable<_Alloc> && allocator_traits<_Alloc>::is_always_equal::value) {
-            // make stateless allocator
+            // recreate stateless allocator
             _Alloc _Al{};
-            const size_t _Count = (_Size + sizeof(_Aligned_block) - 1) / sizeof(_Aligned_block);
-            _Al.deallocate(static_cast<_Aligned_block*>(_Ptr), static_cast<_Alloc_size_type>(_Count));
+            const size_t _Size_in_blocks = _Allocation_block_size(_Size);

```

Comment:
Similarly, should we extract `_Size_in_blocks` here? (There's no opportunity for further unification though, since the nature of how we create `_Al` varies.)

Reply:
I could put the allocator recreation/retrieval in an IILE to eliminate the tail duplication. That normally feels a little too "clever" to me, but it actually seems readable here. I'll push it and see what reviewers think.